src/c_api/value.cpp

```diff
@@ -610,6 +610,36 @@ kuzu_state kuzu_value_get_timestamp_tz(kuzu_value* value, kuzu_timestamp_tz_t* o
     return KuzuSuccess;
 }
 
+kuzu_state kuzu_value_get_decimal_as_string(kuzu_value* value, char** out_result) {
+    auto logical_type_id = static_cast<Value*>(value->_value)->getDataType().getLogicalTypeID();
+    auto physical_type_id = static_cast<Value*>(value->_value)->getDataType().getPhysicalType();
+    if (logical_type_id != LogicalTypeID::DECIMAL) {
+        return KuzuError;
+    }
+    switch (physical_type_id) {
+    case PhysicalTypeID::INT16:
+        *out_result = kuzu_value_to_string(value);
+        return KuzuSuccess;
+
+    case PhysicalTypeID::INT32:
+        *out_result = kuzu_value_to_string(value);
+        return KuzuSuccess;
+
+    case PhysicalTypeID::INT64:
+        *out_result = kuzu_value_to_string(value);
+        return KuzuSuccess;
+
+    case PhysicalTypeID::INT128: {
+        *out_result = kuzu_value_to_string(value);
+        return KuzuSuccess;
+    }
+
+    default:
+        return KuzuError;
+    }
+    return KuzuSuccess;
+}
```

Comment:
You did not change this part as I suggested. 
> You should NOT reinvent the logic of converting decimal to string. There is already a function `toString` you can use.

Reply:
Is there a reason we want to use the `toString` function instead of calling another c_api? the `toString` method will be returning a `const char*` variable which I can't seem to find an easy way of translating into a `char*`, and `kuzu_value_to_string` returns exactly the datatype we want.

I reimplemented this with `toString`, currently just testing locally to see if everything still works.

Reply:
You are working on an API binding, which should not care about the underlying physical type. If a new physical type is added or the underlying core implementation is changed, as long as the same interface function is provided, no changes should be required at the API-level. Your implementation is depending on the underlying physical type, which:

1. May create additional maintenance burden in the future.
2. You are duplicating some logic. I am sure that the `toString` function is already able to handle the type switching you have done.

If you have question about a change I request, you should ask me first. Not doing it and requesting a re-review is unproductive as I have to read the same code twice.